ConsumerApi.Tests.Integration/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -1,119 +1,109 @@
-    private Client _sdk = null!;
-    private bool _isAuthenticated;
-    private readonly HttpClient _httpClient;
-    private readonly ClientCredentials _clientCredentials;
+    private readonly ChallengesContext _challengesContext;
+    private readonly IdentitiesContext _identitiesContext;
+    private readonly ResponseContext _responseContext;
 
-    public ChallengesApiStepDefinitions(IOptions<HttpConfiguration> httpConfiguration, HttpClientFactory factory)
+    public ChallengesApiStepDefinitions(ChallengesContext challengesContext, IdentitiesContext identitiesContext, ResponseContext responseContext)
     {
-        _challengeId = string.Empty;
-        _httpClient = factory.CreateClient();
-        _clientCredentials = new ClientCredentials(httpConfiguration.Value.ClientCredentials.ClientId, httpConfiguration.Value.ClientCredentials.ClientSecret);
+        _challengesContext = challengesContext;
+        _identitiesContext = identitiesContext;
+        _responseContext = responseContext;
     }
 
-    [Given("the user is authenticated")]
-    public async Task GivenTheUserIsAuthenticated()
-    {
-        _sdk = await Client.CreateForNewIdentity(_httpClient, _clientCredentials, Constants.DEVICE_PASSWORD);
-        _isAuthenticated = true;
-    }
+    private Client AnonymousClient => _identitiesContext.AnonymousClient!;
+    private Client Identity(string identityName) => _identitiesContext.Identities[identityName];
+    private string ChallengeId => _challengesContext.ChallengeId!;
+    private ApiResponse<Challenge> ChallengeResponse => _responseContext.ChallengeResponse!;
 
-    [Given("the user is unauthenticated")]
-    public void GivenTheUserIsUnauthenticated()
+    #region Given
+    [Given("a Challenge c created by (.+)")]
+    public async Task GivenAChallengeCreatedByI(string identityName)
     {
-        _sdk = Client.CreateUnauthenticated(_httpClient, _clientCredentials);
-        _isAuthenticated = false;
+        _responseContext.ChallengeResponse = _challengesContext.IsAuthenticated ?
+            await Identity(identityName).Challenges.CreateChallenge() :
+            await Identity(identityName).Challenges.CreateChallengeUnauthenticated();
+        ChallengeResponse.Should().BeASuccess();
+
+        _challengesContext.ChallengeId = ChallengeResponse.Result!.Id;
+        ChallengeId.Should().NotBeNullOrEmpty();
     }
 
-    [Given("a Challenge c")]
+    [Given(@"a Challenge c")]
     public async Task GivenAChallengeC()
     {
-        var challengeResponse = !_isAuthenticated ? await _sdk.Challenges.CreateChallengeUnauthenticated() : await _sdk.Challenges.CreateChallenge();
-        challengeResponse.Should().BeASuccess();
-        _challengeId = challengeResponse.Result!.Id;
+        _responseContext.ChallengeResponse = _challengesContext.IsAuthenticated ?
+            await AnonymousClient.Challenges.CreateChallenge() :
+            await AnonymousClient.Challenges.CreateChallengeUnauthenticated();
```

Comment:
So that basically means that there is no point in calling `CreateChallenge` as anonymous client, as well as `CreateChallengeUnauthenticated` with the authenticated identity?

Reply:
Correct. Doesn't this make sense to you?

Reply:
Of course not, but I would expect an exception to be thrown if someone tries one such "illegal" operation, or at least some kind of indicator that it doesn't make sense.

Reply:
But this would be a test for the SDK, right? So we shouldn't consider this for the tests of the API.

Reply:
No, I was talking about updating the SDK :) This is unrelated to tests.

Reply:
I'll check.